File: src/Lykke.Service.PayInvoicePortal/Models/Invoices/InvoiceModel.cs
using System;
using System.Collections.Generic;

namespace Lykke.Service.PayInvoicePortal.Models.Invoices
{
    public class InvoiceModel
    {
        public string Id { get; set; }
        public string Number { get; set; }
        public string ClientName { get; set; }
        public string ClientEmail { get; set; }
        public double Amount { get; set; }
        public DateTime DueDate { get; set; }
        public string Status { get; set; }
        public string SettlementAsset { get; set; }
        public int SettlementAssetAccuracy { get; set; }
        public string WalletAddress { get; set; }
        public DateTime CreatedDate { get; set; }
        public string Note { get; set; }
        public IReadOnlyList<FileModel> Files { get; set; }
        public IReadOnlyList<HistoryItemModel> History { get; set; }
    }
}

File: src/Lykke.Service.PayInvoicePortal/wwwroot/ng2/src/app/components/InvoiceDetails/InvoiceModel.ts
import { FileModel } from '../../models/FileModel';
import { HistoryItemModel } from './History/HistoryItemModel';

export class InvoiceModel {
  constructor(
    public id: string = '',
    public number: string = '',
    public clientName: string = '',
    public clientEmail: string = '',
    public amount: number = 0,
    public dueDate: Date = null,
    public status: string = '',
    public settlementAsset: string = '',
    public settlementAssetDisplay: string = '',
    public settlementAssetAccuracy: string= '',
    public paymentAsset: string = '',
    public paymentAssetNetwork: string = '',
    public walletAddress: string = '',
    public createdDate: string = '',
    public note: string = ''
  ) {
    this.files = new Array<FileModel>();
    this.history = new Array<HistoryItemModel>();
  }

  public files: FileModel[];
  public history: HistoryItemModel[];
}
